### DISCUSSION TITLE
Using output of one action in another

### DISCUSSION THREAD

**Discussion by tmarrinan** [Alpine-DAV/ascent#1392] [Q&A]
Hello.
I have a situation where I'd like to use the output of one action as data for another action. Here are two use cases:

CASE 1
Action 1: render a scene to an image
Action 2: process the image for further analysis

CASE 2
Action 1: create pipeline to filter simulation data and create mesh
Action 2: use the mesh data in custom Python action

The first case, I'm not sure how to accomplish - it seems that scenes are saved as images to disk. Is there any way to not save to disk, but get a handle on the image data instead?

The second case seems more possible, but I can't tell from the example in the docs whether each rank has a subset of the mesh or if the Python script is only run on rank 0 and contains the whole mesh.

**Comment by cyrush** [comment]
CASE 1 is not generally possible b/c we do render to the file system.

However, if you create a scalar rendering, you get a dataset of the resulting image that looks like a mesh.
https://ascent.readthedocs.io/en/latest/Actions/Examples.html#an-example-of-using-devil-ray-scalar-rendering

You can feed that into other operations (like a python extract)


CASE 2 is possible:

There is a python interpreter on each MPI task, and they will each receive the subset of the domains that exist on that MPI task. 
(There could be tasks w/o data depending on the pipeline operation)
